File: USACO/Contest/2016/USOpen/Bronze/diamond.java
import java.util.*;
import java.io.*;

public class diamond {
  public static void main(String[] args) throws FileNotFoundException {
    Scanner in = new Scanner(new File("diamond.in"));
    int n = in.nextInt();
    int k = in.nextInt();

    int[] diamonds = new int[n];

    for(int i = 0; i<n; i++){
      diamonds[i] = in.nextInt();
    }
    in.close();

    int result = 0;

    for(int i = 0; i<n; i++){
      int count = 0;
      for(int j = 0; j<n; j++){
        if(diamonds[j] >= diamonds[i] && Math.abs(diamonds[j]-diamonds[i]) <= k){
          count += 1;
        }
      }
      result = Math.max(result, count);
    }

    System.out.println("Result: " + String.valueOf(result));

    PrintWriter out = new PrintWriter(new File("diamond.out"));
    out.println(result);
    out.close();
  }
}

File: USACO/Practice/2016/USOpen/Bronze/diamond.py
import string
import math
import itertools
from sys import maxsize


def convert_arr_int(arr):
    return list(map(int, arr))


def main():
    with open("diamond.in", "r") as input_file:
        input_data = input_file.read().split("\n")
    n = int(input_data[0].split(" ")[0])
    k = int(input_data[0].split(" ")[1])

    diamonds = []

    for i in range(n):
        diamonds.append(int(input_data[1+i]))

    max_diamonds = -maxsize

    for i in range(n):
        count = 0
        for j in range(n):
            if diamonds[j] >= diamonds[i] and abs(diamonds[j] - diamonds[i]) <= k:
                count += 1
        max_diamonds = max(max_diamonds, count)

    print("Result:", str(max_diamonds))

    output_file = open("diamond.out", "w")
    output_file.write(str(max_diamonds))
    output_file.close()


if __name__ == "__main__":
    main()
